src/form-builder-controls.js

```diff
@@ -61,41 +62,171 @@ const colorProperty = {
-      content: 'danger',
-    },
-    {
-      value: 'text-warning',
-      content: 'warning',
-    },
-    {
-      value: 'text-info',
-      content: 'info',
-    },
-    {
-      value: 'text-light',
-      content: 'light',
-    },
-    {
-      value: 'text-dark',
-      content: 'dark',
-    },
+    options: [
+      {
+        value: 'text-primary',
+        content: 'primary',
+      },
+      {
+        value: 'text-secondary',
+        content: 'secondary',
+      },
+      {
+        value: 'text-success',
+        content: 'success',
+      },
+      {
+        value: 'text-danger',
+        content: 'danger',
+      },
+      {
+        value: 'text-warning',
+        content: 'warning',
+      },
+      {
+        value: 'text-info',
+        content: 'info',
+      },
+      {
+        value: 'text-light',
+        content: 'light',
+      },
+      {
+        value: 'text-dark',
+        content: 'dark',
+      },
+    ],
+  },
+};
+const KeyName = {
+  type: 'FormInput',
+  field: 'name',
+  panel: 'variable',
+  config: {
+    label: 'Key Name',
+    name: 'Key Name',
+    validation: 'required',
+    helper: 'A variable key name is a symbolic name to reference information.',
+  },
+};
+const DataType = {
+  type: 'FormMultiselect',
+  field: 'dataFormat',
+  panel: 'variable',
+  config: {
+    label: 'Data Type',
+    name: 'Data Type',
+    helper: 'The data type specifies what kind of data is stored in the variable.',
+    validation: 'required',
+    options: [
+      {
+        value: 'string',
+        content: 'Text',
+      },
+      {
+        value: 'int',
+        content: 'Integer',
+      },
+      {
+        value: 'float',
+        content: 'Decimal',
+      },
+      {
+        value: 'datetime',
+        content: 'Datetime',
+      },
+      {
+        value: 'date',
+        content: 'Date',
+      },
+      {
+        value: 'currency',
+        content: 'Currency',
+      },
+      {
+        value: 'boolean',
+        content: 'Boolean',
+      },
+    ],
+  },
+};
+const DataTypeWithoutDate = {
+  type: 'FormMultiselect',
+  field: 'dataFormat',
+  panel: 'variable',
+  config: {
+    label: 'Data Type',
+    name: 'Data Type',
+    helper: 'The data type specifies what kind of data is stored in the variable.',
+    validation: 'required',
+    options: [
+      {
+        value: 'string',
+        content: 'Text',
+      },
+      {
+        value: 'int',
+        content: 'Integer',
+      },
+      {
+        value: 'float',
+        content: 'Decimal',
+      },
+      {
+        value: 'currency',
+        content: 'Currency',
+      },
+      {
+        value: 'boolean',
+        content: 'Boolean',
+      },
     ],
   },
 };
+const ValidateVariable = {
+  type: 'FormInput',
+  field: 'validation',
+  panel: 'variable',
+  config: {
+    label: 'Validation',
+    name: 'validation',
+    helper: 'Validation rules ensure the integrity and validity of the data.',
+  },
+};
+const ReadOnly = {
```

Comment:
I like that this was broken out into a reusable object 👍 

@velkymx @marcoAntonioNina

There is already a PR open for adding the "read only" property: https://github.com/ProcessMaker/screen-builder/pull/265/files. Not sure why this work was duplicated here.

This also seems to set the `disabled` attribute, not the `readonly` attribute. Bootstrap components accept `readonly`: https://getbootstrap.com/docs/4.0/components/forms/#readonly. Was this intentional, or do we actually want this to disable the component? From the perspective of an average user, the difference may to be very noticeable, and `readonly` only seems to work on text input and textarea.

Reply:
Following the requirement in the ticket: https://github.com/ProcessMaker/processmaker/issues/2001

The readonly property must be part of the Variable panel

![image](https://user-images.githubusercontent.com/8028650/60603648-24ca6a80-9d84-11e9-8ec1-f82dccd450db.png)